app/gui/src/project-view/components/visualizations/JSONVisualization/JsonArrayWidget.vue

```diff
@@ -1,10 +1,12 @@
 <script lang="ts" setup>
 import JsonValueWidget from '@/components/visualizations/JSONVisualization/JsonValueWidget.vue'
+import { Opt } from '@/util/data/opt'
 import { computed } from 'vue'
+import { CreateProjection } from './types'
 
-const props = defineProps<{ data: unknown[] }>()
-const emit = defineEmits<{
-  createProjection: [path: (string | number)[][]]
+const props = defineProps<{
+  data: unknown[]
+  onCreateProjection?: Opt<CreateProjection>
```

Comment:
We have found before that naming a property starting with `on` can lead to problems